src/components/ShortyString/ShortyString.tsx

```diff
@@ -27,12 +27,21 @@ export default function ShortyString({
     collapseLabel = 'Show less',
 }: Props) {
     const [expanded, setExpanded] = React.useState(false);
-    const hasToggle = value.length > limit;
-    const length =
-        displayLength && !expanded ? `(${value.length} symbols)` : undefined;
 
-    const text = expanded || value.length <= limit ? value : value.slice(0, limit - 4) + '\u00a0...';
-    const label = expanded ? collapseLabel : expandLabel;
+    const toggleLabelAction = expanded ? collapseLabel : expandLabel;
+    const toggleLabelSymbolsCount = displayLength && !expanded
+        ? ` (${value.length} symbols)`
+        : '';
+    const toggleLabel = toggleLabelAction + toggleLabelSymbolsCount;
+
+    // showing toogle button with a label that is longer than the hidden part is pointless,
+    // hence compare to limit + length
+    const hasToggle = value.length > limit + toggleLabel.length;
+
+    const text = expanded || !hasToggle
+        ? value
```

Comment:
Having set limit for the value to a certain level via props, in fact the length of a shown text will be the limit + the length of the button. For example, in the app there is a place where it will be shown more than 230 symbols instead of limit set to 200. Won’t it be a little misleading?

Reply:
A valid point. The the simplest way out of this is to introduce a prop that would toggle this behavior. Will do that.